src/h5web/visualizations/heatmap/config.ts

```diff
@@ -1,58 +1,47 @@
+import { combine } from 'zustand/middleware';
 import type { ColorMap } from './models';
 import { Domain, ScaleType } from '../shared/models';
 import { StorageConfig, createPersistableState } from '../../storage-utils';
 
-interface HeatmapConfig {
+type HeatmapConfig = {
   dataDomain: Domain | undefined;
-  resetDomains: (dataDomain: Domain | undefined) => void;
-
   requestedDomain: Domain | undefined;
-  setRequestedDomain: (requestedDomain: Domain | undefined) => void;
-
   colorMap: ColorMap;
-  setColorMap: (colorMap: ColorMap) => void;
-
   scaleType: ScaleType;
-  setScaleType: (scaleType: ScaleType) => void;
-
   keepAspectRatio: boolean;
-  toggleAspectRatio: () => void;
-
   showGrid: boolean;
-  toggleGrid: () => void;
-}
+};
```

Comment:
The new `combine` middleware allows splitting the state from the actions. The actions no longer have to be typed.